Lib/test/test_inspect.py

```diff
@@ -2321,6 +2321,138 @@ async def func(a=None):
                          {'a': None, 'gencoro': gencoro, 'b': 'spam'})
 
 
+class TestGetAsyncGenState(unittest.TestCase):
+
```

Comment:
This test would be better if you create the gen in each test separately and use builtin `anext` to iterate over.